tests/tools_tests/ppx/TestPpx.res

```diff
@@ -24,3 +24,8 @@ module Arity = {
   let two = (x, y) => x + y
   let n = two(one(1), 5)
 }
+
+@module("react")
+external useState: (unit => 'state) => string = "useState"
```

Comment:
@cknitt here's the repro, looks like it's a simple AST conversion issue between the legacy and current one.

Reply:
ppx AST conversion was still missing from this PR -- added

Reply:
Great! Regarding the async/await issue I mentioned, while this works now

```rescript
let f = async () => ()

let g = async () => await f()
```

there are other cases that don't, such as

```rescript
let f = async (promise, _x) => await promise
```

which still gives

```
Await on expression not in an async context
```

Reply:
With e.g.

```
"ppx-flags": ["rescript-schema-ppx/bin"],
```

I mean.
Without PPX it works fine.

Reply:
Thanks for catching this @cknitt  -- should be fixed now.

Reply:
I can confirm, everything working fine now! 🎉